deeplearning4j-scaleout/spark/dl4j-spark/src/main/java/org/deeplearning4j/spark/impl/SparkListenable.java

```diff
@@ -0,0 +1,99 @@
+package org.deeplearning4j.spark.impl;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.deeplearning4j.api.storage.StatsStorageRouter;
+import org.deeplearning4j.api.storage.StatsStorageRouterProvider;
+import org.deeplearning4j.api.storage.listener.RoutingIterationListener;
+import org.deeplearning4j.optimize.api.IterationListener;
+import org.deeplearning4j.spark.api.TrainingMaster;
+import org.deeplearning4j.spark.impl.graph.SparkComputationGraph;
+import org.deeplearning4j.spark.impl.listeners.VanillaStatsStorageRouterProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/** Created by huitseeker on 2/15/17. */
+@Slf4j
+public class SparkListenable implements Serializable {
```

Comment:
This shouldn't implement serializable. TrainingMasters aren't serializable (nor should they be), and ~~~instances of these (SparkDl4jMultiLayer and SparkComputationGraph) aren't either.~~~

Edit: turns out SparkDl4jMultiLayer and SparkComputationGraph *are* serializable. But IMO they shouldn't be (again, TrainingMaster instances aren't, and we don't want to use Java serialization for the MultiLayerNetwork and ComputationGraph fields anyway). Implementing serializable suggests to users that java serialization is valid for persisting these objects (it's not).